File: Chapter9_Example/src/main/java/Company.java
import java.util.Objects;

public class Company {
    private final String domainName;
    private int numberOfEmployees;

    public Company(String domainName, int numberOfEmployees) {
        this.domainName = domainName;
        this.numberOfEmployees = numberOfEmployees;
    }

    public String getDomainName() {
        return domainName;
    }

    public int getNumberOfEmployees() {
        return numberOfEmployees;
    }

    public void changeNumberOfEmployees(int delta) {
        assert numberOfEmployees + delta >= 0;
        numberOfEmployees += delta;
    }

    public boolean isEmailCorporate(String email) {
        String emailDomain = email.split("@")[1];
        return Objects.equals(emailDomain, domainName);
    }
}

File: unit-testing/src/main/kotlin/book/chapter10/transaction/Company.kt
package book.chapter10.transaction

class Company(domainName: String, numberOfEmployees: Int) {

    var domainName: String = domainName
        private set
    var numberOfEmployees: Int = numberOfEmployees
        private set

    fun changeNumberOfEmployees(delta: Int) {
        require(numberOfEmployees + delta >= 0)
        numberOfEmployees += delta
    }

    fun isEmailCorporate(email: String): Boolean {
        val emailDomain = email.split('@')[1]
        return emailDomain == domainName
    }

}